src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,48 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final int DATE_INDEX = 0;
+    private static final int NAME_INDEX = 1;
+    private static final int HOUR_INDEX = 2;
+    private static final int SALARY_PER_HOUR_INDEX = 3;
+
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder summaryReport = new StringBuilder("Report for period ");
+        summaryReport
+                .append(dateFrom)
+                .append(" - ")
+                .append(dateTo);
+        LocalDate startDate = parseDate(dateFrom);
+        LocalDate endDate = parseDate(dateTo);
+        for (int i = 0; i < names.length; i++) {
+            summaryReport.append("\n");
```

Comment:
Consider using System.lineSeparator() instead of '\n' for line separation. This ensures compatibility across different operating systems as mentioned in the checklist.